react-app/src/components/Opensearch/main/Filtering/index.tsx

```diff
@@ -30,8 +31,13 @@ export const OsFiltering: FC<{
           disabled={!!props.disabled}
         />
         <div className="flex justify-center flex-row gap-2">
-          <OsExportData columns={props.columns} disabled={context?.data?.total.value === 0}/>
+          <VisibilityPopover
+            list={props.columns.filter((COL) => !COL.locked || COL.field)}
+            onItemClick={props.onToggle}
+            hiddenColumns={props.columns.filter((COL) => COL.hidden === true)}
```

Comment:
What's COL? Is it `col`? 

Also minor nitpicking here: you use a shorthand on line 35 but an explicit check on line 37. I personally prefer explicit, but you do what you prefer.

Reply:
I just realized that COL is being used everywhere. If you feel like we should be consistent, ignore my comment.

Reply:
I meant:
```
// implicit check
!value
```
vs.
```
// explicit check
value === boolean
```